### PR TITLE
Metadata docs

### PR DESCRIPTION
**Pull request by perolavsvendsen**
Add documentation of the data model to the fmu-dataio docs.

### CODE REVIEW

File: docs/datamodel.rst [hunk 1]
```diff
@@ -0,0 +1,624 @@
+The FMU results data model
+##########################
+
+This section described the data model used for FMU results when exporting with
```
Comment (jcrivenaes) [inline docs/datamodel.rst:4]: describes

File: docs/datamodel.rst [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+created.
+
+About the data model
+===================
+
+Why is it made?
+---------------
+
+FMU is a mighty system developed by and for the subsurface community in Equinor, to
+make reservoir modeling more effective, less error-prone and more repeatable with higher quality,
```
Comment (jcrivenaes) [inline docs/datamodel.rst:42]: effective -> efficient ?

**Review by jcrivenaes — APPROVED** [review]
Except from a few typos I have found and potentially some I have missed, I think we just should merge it in :rocket:

File: docs/datamodel.rst (hunk 2 above)
Comment (perolavsvendsen) [reply docs/datamodel.rst]: Fixed

File: docs/datamodel.rst (hunk 1 above)
Comment (perolavsvendsen) [reply docs/datamodel.rst]: Fixed

**Pull request merged**